Fix Engine::insertLibrary() and add a test for it

buschmann23 opened the pull request:
Inserting a library directly with the new insertLibrary() function added the library to the list of already loaded libraries (EnginePrivate::m_libraries), but the Parser uses the return value of Engine::defaultLibraries() to load the tags and filters of each library.

https://github.com/cutelyst/cutelee/blob/33b629a6af6168a5eedd1be79b90cb3187b35a57/templates/lib/parser.cpp#L90-L109

So the tag or filter from the inserted library can not be found by the parser and it throws an error like: `Error while rendering template: Invalid block tag on line 0: 'my_nice_tag''. Did you forget to register or load this tag?`

Adding the name of the inserted library to the list of default libraries (EnginePrivate::m_defaultLibraries) fixes this issue.

I think that should be fine as Engine::loadDefaultLibraries() and Engine::loadLibrary() both check if the library has already been loaded. We might could change the name of insertLibrary() to
insertDefaultLibrary() to make it more clear that it also populates the list of default libraries.

I would like you to review it. Is only a one liner, but your understanding of the code might be better than mine. :)

dantti commented:
Yes, I noted when I tried to use it, in fact we would need a {% load %} tag to load it, so IMHO we could add the insertDefaultLibrary which also adds to the default plugins.

buschmann23 commented:
> Yes, I noted when I tried to use it, in fact we would need a {% load %} tag to load it, so IMHO we could add the insertDefaultLibrary which also adds to the default plugins.

There is a {% load %} tag for plugin libraries.

So, should I rename it to insertDefaultLibrary() ?

dantti commented:
Yes, I know about the load, what I mean is that if the user want's to use the plugin with the current function he uses load tag, or insertDefaultLibrary which will load them all...

buschmann23 commented:
So that the class is dynamically instantiated through meta object system?

dantti commented:
> So that the class is dynamically instantiated through meta object system?

no, so that it's tags are only available if loaded

buschmann23 commented:
I think I now got it. At least I hope so. :D

I now added a separate function for insertDefaultLibrary() and added a test for loading inserted (not default) libraries.

dantti commented:
awesome thanks
